components/infobox/wikis/stormgate/infobox_skill_custom.lua

```diff
@@ -0,0 +1,245 @@
+	if self.faction then
+		table.insert(categories, self.faction .. ' ' .. args.informationType)
+	end
+
+	return categories
+end
+
+---@param lpdbData table
+---@param args table
+---@return table
+function CustomSkill:addToLpdb(lpdbData, args)
+	lpdbData.information = self.faction
+	lpdbData.extradata = {
+		luminite = tonumber(args.luminite),
+		totalluminite = tonumber(args.totalluminite),
+		therium = tonumber(args.therium),
+		totaltherium = tonumber(args.totaltherium),
+		buildtime = tonumber(args.buildtime),
+		totalbuildtime = tonumber(args.totalbuildtime),
+		animus = tonumber(args.animus),
+		totalanimus = tonumber(args.totalanimus),
+		techrequirements = self:_readCommaSeparatedList(args.tech_requirement),
+		buildingrequirements = self:_readCommaSeparatedList(args.building_requirement),
+		targets = self:_readCommaSeparatedList(args.target),
+		casters = self:getAllArgsForBase(args, 'caster'),
+		hotkey = args.hotkey,
+		hotkey2 = args.hotkey2,
+		macrokey = args.macro_key,
+		macrokey2 = args.macro_key2,
+		energy = tonumber(args.energy),
+		duration = tonumber(args.duration),
+		from = args.from,
+		upgradetarget = self:_readCommaSeparatedList(args.upgrade_target),
+		range = tonumber(args.range),
+		radius = tonumber(args.radius),
+		cooldown = tonumber(args.cooldown),
+		castingtime = tonumber(args.casting_time),
+		unlocks = self:_readCommaSeparatedList(args.unlocks),
+		effect = self:_readCommaSeparatedList(args.effect),
+		trigger = args.trigger,
+		invulnerable = self:_readCommaSeparatedList(args.invulnerable),
+		damage = args.damage,
+		damagepercentage = args.damage_percentage,
+		damagetotal = args.damageTotal,
+		damagedps = args.damageDps,
+		damageovertime = args.damage_over_time,
+		health = args.health,
+		healthpercentage = args.health_percentage,
+		healthtotal = args.healthTotal,
+		healthdps = args.healthDps,
+		healthovertime = args.health_over_time,
+		specialcost = args.special_cost
+	}
+
+	return lpdbData
+end
+
+function CustomSkill:_costDisplay()
+	local args = self.args
+	local energy = tonumber(args.energy) or 0
+
+	return table.concat(Array.append({},
+		CostDisplay.run{
+			faction = self.faction,
+			luminite = args.luminite,
+			luminiteTotal = args.totalluminite,
+			therium = args.therium,
+			theriumTotal = args.totaltherium,
+			animus = args.animus,
+			animusTotal = args.totalanimus,
+			buildTime = args.buildtime,
+			buildTimeTotal = args.totalbuildtime,
+		},
+		energy ~= 0 and (ENERGY_ICON .. '&nbsp;' .. energy) or nil,
+		args.special_cost
+	), '&nbsp;')
+end
+
+---@param inputString string?
+---@param makeLink boolean?
+---@return string[]
+function CustomSkill:_readCommaSeparatedList(inputString, makeLink)
+	if String.isEmpty(inputString) then return {} end
+	---@cast inputString -nil
+	local values = Array.map(mw.text.split(inputString, ','), String.trim)
+	if not makeLink then return values end
+	return Array.map(values, function(value) return Page.makeInternalLink(value) end)
+end
```

Comment:
Should we extract this somewhere, i.e. Infobox/Base or Array or something?

Reply:
if we extract imo array

Reply:
#3985